charts/cluster-api-runtime-extensions-nutanix/addons/cni/cilium/values-template.yaml

```diff
@@ -33,11 +33,7 @@ socketLB:
 envoy:
   image:
     useDigest: false
-{{- with .ControlPlane }}
-{{- range $key, $val := .metadata.annotations }}
-{{- if eq $key "controlplane.cluster.x-k8s.io/skip-kube-proxy" }}
 k8sServiceHost: auto
-kubeProxyReplacement: true{{ break }}
-{{- end }}
-{{- end }}
+{{- if .EnableKubeProxyReplacement }}
```

Comment:
Using builtin topolgy templating results in a deadlock because the .ControlPlane is not updated until after the lifecycle hook returns.